storage/replica_command.go

```diff
@@ -1359,6 +1359,18 @@ func (r *Replica) splitTrigger(batch engine.Engine, split *roachpb.SplitTrigger)
 			// Our in-memory state has diverged from the on-disk state.
 			log.Fatalf("failed to update Store after split: %s", err)
 		}
+
+		// To avoid leaving the new range unavailable as it waits to elect
+		// its leader, one (and only one) of the nodes should start an
+		// election as soon as the split is processed. For simplicity, we
+		// choose the first node in the replica list. If this node is
+		// unavailable, the group will have to wait for an election
+		// timeout, just as with any other leader failure. (we could
+		// improve this by e.g. choosing the node that had the leader
+		// lease before the split and is therefore known to be up)
+		if r.store.StoreID() == split.NewDesc.Replicas[0].StoreID {
+			r.store.multiraft.Campaign(split.NewDesc.RangeID)
```

Comment:
Ah, but there is still a problem here. An uninitialized replica can vote, but when the replica becomes initialized the raft group is deleted and recreated, discarding any pending messages. It's a rare race (for it to matter *both* followers need to drop their vote messages), so it might be OK to just wait for an election timeout when it happens, but in this new test that's not an option and so the test fails about 1 in 200 times.